kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/balance/view/BalanceView.java

```diff
@@ -1,120 +1,260 @@
+		inflate(context, R.layout.kinecosystem_balance_view, this);
+		setLayoutParams(new ConstraintLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT,
+			ViewGroup.LayoutParams.WRAP_CONTENT));
+		int topPadding = getResources().getDimensionPixelOffset(R.dimen.kinecosystem_main_medium_margin);
+		setPadding(0, topPadding, 0, 0);
+		initColors();
+		initSizes();
+
+		TypedArray styledAttributes = context.getTheme()
+			.obtainStyledAttributes(attrs, R.styleable.KinEcosystemBalanceView, 0, 0);
+		String subtitle;
+		try {
+			subtitle = styledAttributes.getString(R.styleable.KinEcosystemBalanceView_subTitle);
+
+		} finally {
+			styledAttributes.recycle();
+		}
+
+		this.subTitleTextView = findViewById(R.id.sub_title);
+		this.balanceText = findViewById(R.id.balance_text);
+		this.arrow = findViewById(R.id.arrow);
+		addBalanceViewFactory();
+		addSubTitleViewFactory();
+		setOnClickListener(new OnClickListener() {
+			@Override
+			public void onClick(View v) {
+				balancePresenter.balanceClicked();
+			}
+		});
+		this.arrow.getViewTreeObserver().addOnGlobalLayoutListener(new OnGlobalLayoutListener() {
+			@Override
+			public void onGlobalLayout() {
+				if (arrow.getMeasuredWidth() > 0) {
+					arrowFromX = arrow.getPivotX() - getResources().getDimensionPixelOffset(R.dimen.kinecosystem_main_medium_margin);
+					arrowToX = arrowFromX + arrow.getWidth();
+					arrow.getViewTreeObserver().removeOnGlobalLayoutListener(this);
+				}
+			}
+		});
+
+		this.subTitleTextView.setText(subtitle);
+	}
+
+	private void initColors() {
+		blueColor = ContextCompat.getColor(getContext(), R.color.kinecosystem_bluePrimary);
+		orangeColor = ContextCompat.getColor(getContext(), R.color.kinecosystem_orange);
+		redColor = ContextCompat.getColor(getContext(), R.color.kinecosystem_red);
+		subTitleColor = ContextCompat.getColor(getContext(), R.color.kinecosystem_subtitle);
+	}
+
+	private void initSizes() {
+		Resources resources = getResources();
+		balanceTextSize = resources.getDimension(R.dimen.kinecosystem_title_xlarge_size);
+		subTitleTextSize = resources.getDimension(R.dimen.kinecosystem_sub_title_size);
+	}
+
+
+	private void addBalanceViewFactory() {
+		balanceText.setFactory(new ViewFactory() {
+			@Override
+			public View makeView() {
+				TextView balanceText = new TextView(getContext());
+				balanceText.setTextSize(TypedValue.COMPLEX_UNIT_PX, balanceTextSize);
+				balanceText.setTextColor(blueColor);
+				return balanceText;
+			}
+		});
+	}
+
+	private void addSubTitleViewFactory() {
+		subTitleTextView.setFactory(new ViewFactory() {
+			@Override
+			public View makeView() {
+				TextView subTitle = new TextView(getContext());
+				subTitle.setTextSize(TypedValue.COMPLEX_UNIT_PX, subTitleTextSize);
+				subTitle.setTextColor(subTitleColor);
+				return subTitle;
+			}
+		});
+	}
+
+	@Override
+	public void animateArrow(boolean showArrow) {
+		if (showArrow) {
+			postOnAnimation(new Runnable() {
```

Comment:
just to double check - postOnAnimation is safe to be called from non UI thread right?

Reply:
yes